core/src/main/java/com/adobe/guides/aem/components/core/models/GuidesNavigationImpl.java

```diff
@@ -128,15 +132,36 @@ public List<String> getGuidesNavigation() {
     public String getCurrentPageTocIndex() {
         return currentPageIndexInToc;
     }
+
     @Override
     public String getLimit() {
         return limit;
     }
+
     @Override
     public String getLoadMoreText() {
         return loadMoreText;
     }
 
+    @Override
+    public String getCategoryPath() {
```

Comment:
what if there is no category page at all, it could be the case that the customer has created their own template and didn't add the id=category-page. We should terminate the loop if page.getParent(); return a null or whatever it does when there is no parent anymore.